src/content/docs/cheerp/guides/clientlib.md

```diff
@@ -0,0 +1,327 @@
+
+One example of this is the `get_buffer` function on typed arrays. This
+function used to return `client::ArrayBuffer*`, but it now returns
+`client::_Union<client::ArrayBuffer*, client::SharedArrayBuffer*>*`.
+
+```
+error: cannot initialize a variable of type 'client::ArrayBuffer *' with an rvalue of type '_Union<ArrayBuffer *, SharedArrayBuffer *> *'
+    client::ArrayBuffer* buffer = array->get_buffer();
+                         ^        ~~~~~~~~~~~~~~~~~~~
+```
+
+To fix this, simply cast the union type using `->cast<T>()`. Depending on where
+the result is being used, it may be possible to omit the type parameter to the
+`cast` method.
+
+For example, for the above error, we could instead write:
+
+```cpp
+    client::ArrayBuffer* buffer = array->get_buffer()->cast();
+```
+
+## No known conversion to `ArrayLike<double>*`
+
+Functions that take `ArrayLike<T>` used to be defined as template functions, so
+they could take any type of parameter. These functions are no longer template
+functions. The new `client::ArrayLike<T>` template class is used instead.
+`client::TArray<T>` and typed arrays do not inherit from
+`client::ArrayLike<T>`.
+
+One example of this is the `set` function on typed arrays.
+
+```
+error: no matching member function for call to 'set'
+    array->set(array);
+    ~~~~~~~^~~
+note: candidate function not viable: no known conversion from 'client::Uint8Array *' to 'ArrayLike<double> *' for 1st argument
+        void set(ArrayLike<double>* array);
+             ^
+note: candidate function not viable: requires 2 arguments, but 1 was provided
+        void set(ArrayLike<double>* array, double offset);
+             ^
+```
+
+To pass an array to a function that expects `client::ArrayLike<T>*`, you can
+use `->cast<T>()` or `->cast()`.
+
+```cpp
+    array->set(array->cast());
+```
+
+## Functions with different template parameters
+
+Functions whose return type is not known at compile time were defined as
+template functions. Some of these functions are no longer template functions,
+and return a specific type instead.
+
+One example of this is the `client::JSON.parse` function. This function
+now returns `client::Object*`. The error you will get is slightly confusing.
+
+```
+error: expected '(' for function-style cast or type construction
+    client::Object* object = client::JSON.parse<client::Object*>(string);
+                                                ~~~~~~~~~~~~~~^
+error: expected expression
+    client::Object* object = client::JSON.parse<client::Object*>(string);
+                                                               ^
+```
+
+To fix the error, you can simply remove the type parameter. If you need a type
+that is more specific than what the function returns, you can use `->cast<T>()`
+or `->cast()`.
+
+```cpp
+    client::Object* object = client::JSON.parse(string);
+```
+
+## Redefinition of client class
+
+The old clientlib was missing some modern javascript types. To use these types,
```

Comment:
See Vale's comments (particularly "javascript"). We should decide on a proper casing for clientlib. Is it "clientlib" or "Clientlib", or what?